src/charm.py

```diff
@@ -701,33 +701,63 @@ def _on_add_admin_user_action(self, event: ActionEvent) -> None:
         Args:
             event: Event triggering the add_admin_user action.
         """
-        email = event.params["email"]
-        password = event.params["password"]
         container = self.unit.get_container(CONTAINER_NAME)
-        if container.can_connect():
-            process = container.exec(
-                [
-                    os.path.join(DISCOURSE_PATH, "bin/bundle"),
-                    "exec",
-                    "rake",
-                    "admin:create",
-                ],
-                stdin=f"{email}\n{password}\n{password}\nY\n",
-                working_dir=DISCOURSE_PATH,
-                user=CONTAINER_APP_USERNAME,
-                environment=self._create_discourse_environment_settings(),
-                timeout=60,
+
+        email = event.params["email"]
+
+        if not container.can_connect():
+            event.fail("Unable to connect to container, container is not ready")
+            return
+        self._create_user(event)
```

Comment:
If the create_user fails, it continues running this code. Does not sound like correct.